src/main/java/com/hotels/beans/cache/CacheManagerFactory.java

```diff
@@ -44,11 +42,6 @@
      */
     public static CacheManager getCacheManager(final String cacheName) {
         notNull(cacheName, "cacheName cannot be null!");
-        Map<String, Object> cacheMap = CACHE_MAP.get(cacheName);
-        if (isNull(cacheMap)) {
-            cacheMap = new ConcurrentHashMap<>();
-            CACHE_MAP.put(cacheName, cacheMap);
-        }
-        return new CacheManager(cacheMap);
+        return new CacheManager(CACHE_MAP.computeIfAbsent(cacheName, k -> new ConcurrentHashMap<>()));
```

Comment:
Could simplify the lambda with ConcurrentHashMap::new

Reply:
Yeah, it's because the required function accepts one parameter, while to use the constructor reference it will have to be a Supplier (zero arguments).